internal/directory/okta/okta.go

```diff
@@ -145,7 +157,61 @@ func (p *Provider) UserGroups(ctx context.Context) ([]*directory.Group, []*direc
 	return groups, users, nil
 }
 
+func (p *Provider) getUpdatedGroup(ctx context.Context) ([]*directory.Group, error) {
+	now := time.Now()
+	if p.lastUpdated == nil {
+		p.lastUpdated = &now
+	}
+
+	u := &url.URL{Path: "/api/v1/groups"}
+	q := u.Query()
+	q.Set("limit", strconv.Itoa(p.cfg.batchSize))
+	q.Set("filter", fmt.Sprintf(`lastUpdated+gt+"%[1]s"+or+lastMembershipUpdated+gt+"%[1]s"`, p.lastUpdated.UTC().Format(filterDateFormat)))
```

Comment:
is this the only difference between `getUpdatedGroup` and `getGroups`? Maybe it would be better to just only set this parameter if `p.lastUpdated` is not nil? Then the first time we would just retrieve all the groups?

Reply:
Updated.